src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -128,12 +128,12 @@ public static WorldProvider getProvider(ResourceLocation dimID)
             allWorlds.removeAll(worlds.values());
             for (ListIterator<World> li = allWorlds.listIterator(); li.hasNext(); )
             {
-                World w = li.next();
-                leakedWorlds.add(System.identityHashCode(w));
```

Comment:
The entire point of leak detection is to use the identity hash of the world object, this breaks the entire leak detection system.